Decisions/scs-xxxx-v1-ssd-flavors.md

```diff
@@ -0,0 +1,250 @@
+title: SSD Flavors
+type: Decision Record
+status: Draft
+track: IaaS
+version: v0.1
+enhances: scs-xxxx-v1-flavor-naming.md
+
+# Introduction
+
+SCS defines an IaaS Flavor Naming standard that mandates a number of standard flavors
+to be available in each SCS-compliant IaaS offering. While offering or exposing
+IaaS is not a requirement for SCS-compliant infrastructure offerings -- SCS allows
+for platforms only exposing the container layer (plus S3 compatible object storage)
+for gen2 (container-based) cloud-native workloads --
+the SCS reference implementation does include a complete IaaS implementation that
+many providers want to expose as they have customers desiring access at this layer
+for gen1 (VM-based) cloud-native workloads or for the virtualization of more
```

Comment:
Personally, I'd avoid categorising use cases in this much detail, as does not add too much value to the introduction and may be disputable.

Especially `gen1 (VM-based) cloud-native` strongly implies that Kubernetes is "gen2" cloud native, which might be correct, but definitely is kind of opinionated.
For example, someone might opt to avoid Kubernetes and instead build a cloud native setup by fully embracing the Hashicorp stack which in many ways kind of competes with Kubernetes: Packer for building images, Terraform for VM/resource provisioning, Nomad for workload scheduling, Consul Connect for service meshing etc.
They might not agree that this setup is a "gen1" (hence kind of legacy) setup, it's just different.

Of course, in my impression, using Kubernetes (or Kubernetes with some Hashicorp tools integrated) is the more popular choice - and also the choice SCS settled on as primary use case. Nevertheless, it might be better to focus only on the technological terms, here.

Reply:
Well, the discussion tends to be different: The k8s community considers k8s as the one and only "cloud-native" approach, which is untrue, IMVHO. Cloud-native existed before containers were widespread, but the k8s folks want to monopolize the wording. They even created a foundation with that claim.
So the wording acknowledges that a lot of current ("gen-2") cloud-native development targets containers, while other and older approaches exist ("gen-1") and continue to be valid and used.
There is no rating involved here, merely the observation that we have seen two big waves in the industry here ... the 2nd is larger and more modern.

Reply:
Ah, ok. Over-interpreted the "gen" as rating, then :). Still, even when "gen" only is meant to be understood as hint regarding temporal order or history, such high level categorization in waves/generations still opens a lot of room for potential misunderstandings/discussion.

For example, Nomad was released first one year *after* Kubernetes' first release and is still advertised as alternative. Should it be considered gen-1 cloud-native, even it is "younger" than gen-2? Or gen-3?
What does gen-1 mean exactly in contrast to classical VM workload?

So, I think referring only to well-established high level terms like IaaS/... as well as concrete technical terms like VM/container would eliminate any such potential for misunderstandings or discussions unrelated to the actual topic of the decision record (like the one I just started).

Reply:
OK, ok, I used the words wave 1 and wave 2 instead.